src/common/telemetry/collect-telemetry/collect-telemetry.ts

```diff
@@ -0,0 +1,149 @@
+import {execSync} from 'child_process'
+import * as fs from 'fs'
+import fetch from 'node-fetch'
+import * as path from 'path'
+import {ProjenrcFile} from 'projen'
+import type {NodeProject, NodeProjectOptions} from 'projen/lib/javascript'
+import type {IWithTelemetryReportUrl} from '../with-telemetry'
+import {cloneWorkflow} from './clone-workflow'
+import {diff} from './diff'
+import {EscapeHatches, getEscapeHatches} from './get-escape-hatches'
+
+interface TelemetryPayload {
+  version?: string
+  gitUrls?: Array<string>
+  escapeHatches?: {
+    files?: EscapeHatches
+    overrides?: EscapeHatches
+    tasks?: EscapeHatches
+  }
+  workflows?: {
+    deleted?: Array<string>
+    new?: Array<object>
+    updated?: Array<object>
+  }
+  errors?: unknown
+}
+
+export const telemetryEnableEnvVar = 'IS_OTTOFELLER_TEMPLATES_TELEMETRY_COLLECTED' as const
+
+export const collectTelemetry = async (project: NodeProject & IWithTelemetryReportUrl) => {
+  if (!project.telemetryReportUrl || !process.env[telemetryEnableEnvVar]) {
+    return
+  }
+
+  project.logger.info('Collect project telemetry')
+  const payload: TelemetryPayload = {}
+  const errors: Array<unknown> = []
+
+  // ANCHOR template version
+  try {
+    const version = project.package.tryResolveDependencyVersion('@ottofeller/templates')
+
+    if (version) {
+      payload.version = version
+    }
+  } catch (e) {
+    errors.push(e)
+  }
+
+  // ANCHOR git urls
+  try {
+    const gitUrls = execSync('git remote -v', {encoding: 'utf-8'})
+      .split('\n')
+      .map((item) => item.split(/\s+/)[1])
+
+    const uniqueGitUrls = Array.from(new Set(gitUrls)).filter(Boolean)
+
+    if (uniqueGitUrls.length > 0) {
+      payload.gitUrls = uniqueGitUrls
+    }
+  } catch (e) {
+    errors.push(e)
+  }
+
+  // ANCHOR escape hatches
+  try {
+    const rcFile = ProjenrcFile.of(project)?.filePath
+
+    if (!rcFile) {
+      throw 'projenrc file not found'
+    }
+
+    const rcFilePath = path.resolve(project.outdir, rcFile)
+    const projenrcContents = fs.readFileSync(rcFilePath, 'utf-8')
+    const fileHandles = ['tryFindFile', 'tryFindObjectFile', 'tryRemoveFile', 'tryFindWorkflow']
+    const files = getEscapeHatches(projenrcContents, fileHandles)
+    const overrideHandles = ['addOverride', 'addDeletionOverride', 'addToArray', 'patch']
+    const overrides = getEscapeHatches(projenrcContents, overrideHandles)
+    const taskHandles = ['tryFind', 'addTask', 'removeTask', 'addScripts', 'removeScript']
+    const tasks = getEscapeHatches(projenrcContents, taskHandles)
+
+    if (files || overrides || tasks) {
```

Comment:
@quesabe would this catch only changes in `.projenrc.ts` or changes from the base template as well?

Reply:
Yes, it only finds the calls in projenrc file. We don't get any changes that come when the template package version is updated.